src/main/java/com/iexec/worker/tee/scone/TeeSconeService.java

```diff
@@ -45,112 +45,81 @@ public class TeeSconeService {
+        this.lasServicesManager = lasServicesManager;
+
+        if (isTeeEnabled()) {
             log.info("Worker can run TEE tasks");
         } else {
             LoggingUtils.printHighlightedMessage("Worker will not run TEE tasks");
         }
     }
 
     public boolean isTeeEnabled() {
-        return isLasStarted;
-    }
-
-    boolean startLasService() {
-        String lasImage = sconeConfig.getLasImageUri();
-        DockerRunRequest dockerRunRequest = DockerRunRequest.builder()
-                .containerName(sconeConfig.getLasContainerName())
-                .imageUri(lasImage)
-                // application & post-compose enclaves will be
-                // able to talk to the LAS via this network
-                .dockerNetwork(workerConfigService.getDockerNetworkName())
-                .sgxDriverMode(sgxService.getSgxDriverMode())
-                .maxExecutionTime(0)
-                .build();
-        if (!lasImage.contains(sconeConfig.getRegistryName())) {
-            throw new RuntimeException(String.format("LAS image (%s) is not " +
-                    "from a known registry (%s)", lasImage, sconeConfig.getRegistryName()));
-        }
-        DockerClientInstance client;
-        try {
-            client = dockerService.getClient(
-                    sconeConfig.getRegistryName(),
-                    sconeConfig.getRegistryUsername(),
-                    sconeConfig.getRegistryPassword());
-        } catch (Exception e) {
-            log.error("", e);
-            throw new RuntimeException("Failed to get Docker authenticated client to run LAS");
-        }
-        if (client == null) {
-            log.error("Docker client with credentials is required to enable TEE support");
-            return false;
-        }
-        if (!client.pullImage(lasImage)) {
-            log.error("Failed to download LAS image");
-            return false;
-        }
-
-        DockerRunResponse dockerRunResponse = dockerService.run(dockerRunRequest);
-        if (!dockerRunResponse.isSuccessful()) {
-            log.error("Failed to start LAS service");
-            return false;
-        }
-        return true;
+        return sgxService.isSgxEnabled();
     }
 
+    @Override
     public List<String> buildPreComputeDockerEnv(
-            @Nonnull TeeSessionGenerationResponse session,
-            long heapSize) {
+            TaskDescription taskDescription,
+            @Nonnull TeeSessionGenerationResponse session) {
         String sconeConfigId = session.getSessionId() + "/pre-compute";
-        return getDockerEnv(sconeConfigId, heapSize, session.getSecretProvisioningUrl());
+        String chainTaskId = taskDescription.getChainTaskId();
+        TeeWorkflowConfiguration teeWorkflowConfig =
+                teeWorkflowConfigurationService.getTeeWorkflowConfiguration(chainTaskId);
+        return getDockerEnv(chainTaskId, sconeConfigId, teeWorkflowConfig.getPreComputeHeapSize(), session.getSecretProvisioningUrl());
```

Comment:
FYI: Me might stop overriding heap size in the future

Reply:
Maybe add a TODO in comment?

Reply:
Sure. It could be removed today, but I don't think it would fit into this PR. Please see the new TODO here: https://github.com/iExecBlockchainComputing/iexec-worker/pull/464/commits/984ae97e3e90ba5b3ab1d0101de261202f514d78.